### PR TITLE
Improve nodata handling in zonal

### LINKED ISSUE #22: Improve nodata handling in `do_mean`
The current way to handle metadata in `do_mean`:

https://github.com/WFP-VAM/seasmon_xr/blob/1836cded9ce22f196193bb79b758de3133cb5c59/seasmon_xr/ops/zonal.py#L35

only considers the nodata of the actual input data. The nodata flag for zones is not considered, other than negative values are not allowed.

This can result in the edge case where we have data in the input, but a `0` as zone which is the "nodata" zone in this particular edge case. When that happens, a mean is actually calculated for the nodata zone. 

Should we better exclude such an edge case, handle it outside or just ignore it?

@Kirill888

### PR DESCRIPTION
**Pull request by valpesendorfer**
This PR includes an argument for a zonal nodata value in `do_mean` and the respective accessor, plus additional tests. This covers an edge case where we have valid data but no valid zone. 

**Note**:  The tests already use the `.hdc` accessor root, which is compatible with changes coming in with #21 

This closes #22

### CODE REVIEW

**Comment by valpesendorfer** [comment]
when possible, please review @Kirill888

File: seasmon_xr/ops/zonal.py [hunk 1]
```diff
@@ -32,7 +32,7 @@ def do_mean(pixels, z_pixels, nodata, num_zones, dtype="float64"):
             for cl in range(nc):
                 pix = pixels[tix, rw, cl]
                 z_idx = z_pixels[rw, cl]
-                if pix != nodata and z_idx >= 0:
+                if (pix != nodata) and (z_idx != z_nodata):
```
Comment (Kirill888) [inline seasmon_xr/ops/zonal.py:35]: I guess this one assumes non-float data on input? 
Will need nan check for float32 inputs, but this will be probably be better off with a separate function for float data with appropriate routing in python

**Review by Kirill888 — APPROVED** [review]

File: seasmon_xr/ops/zonal.py (hunk 1 above)
Comment (valpesendorfer) [reply seasmon_xr/ops/zonal.py:35]: Yes, it assumes zones as ints. Could we not add `!= np.nan` as an `or` condition in the second bracket, or are there unintended results if data comes in integer?

Comment (Kirill888) [reply seasmon_xr/ops/zonal.py:35]: I mean for `pix`. 

And no, you can't compare with `nan` directly, it's not equal to anything including itself. For `float32` data instead of `pix != nodata` one would have to have `not isnan(pix)`. But since `nodata` is a compulsory part of the interface the assumption must be that it's using int values.

Comment (valpesendorfer) [reply seasmon_xr/ops/zonal.py:35]: gotcha, yes that'll need to be handled better

File: seasmon_xr/accessors.py [hunk 2]
```diff
@@ -557,6 +557,12 @@ def mean(
         if not isinstance(zones, xarray.DataArray):
             raise ValueError("Zones need to be xarray.DataArray!")
 
+        if "nodata" not in zones.attrs:
+            raise ValueError("Zones xarray DataArray needs nodata attribute")
+
+        # set null values to nodata value
+        xx = xx.where(~xx.isnull(), xx.nodata).astype(xx.dtype)
```
Comment (valpesendorfer) [inline seasmon_xr/accessors.py:564]: @Kirill888 would this suffice to account for `nan` in float input? The `nodata` attribute is required above.

Wasn't sure if we need to call `astype`, but I assumed this would keep int data in int after `where`

Comment (valpesendorfer) [reply seasmon_xr/accessors.py:564]: @Kirill888 even though the PR is approved, please check above

Comment (Kirill888) [reply seasmon_xr/accessors.py:564]: This looks fine, but I think xarray keeps original `dtype`, so not sure if `.astype` is needed here

Comment (Kirill888) [reply seasmon_xr/accessors.py:564]: If it does not then maybe need to wrap `xx.dtype.type(xx.nodata)`

Comment (Kirill888) [reply seasmon_xr/accessors.py:564]: actually `xx.notnull()` is probably more performant than `~xx.isnull()`

**Comment by valpesendorfer** [comment]
implemented latest suggestions @Kirill888 . Going to merge if tests pass
